src/hooks/usePendingActions.ts

```diff
@@ -17,16 +17,16 @@ const getSignableCount = (queue: TransactionListPage, walletAddress: string): nu
   return queue.results.filter((tx) => isTransactionListItem(tx) && isSignableBy(tx.transaction, walletAddress)).length
 }
 
-const usePendingActions = (chainId: string, safeAddress?: string): PendingActions => {
+const usePendingActions = (chainId: string, isVisible: boolean, safeAddress?: string): PendingActions => {
   const wallet = useWallet()
   const { safeAddress: currentSafeAddress } = useSafeInfo()
   const { page: currentSafeQueue } = useTxQueue()
   const isCurrentSafe = currentSafeAddress === safeAddress
 
   const [loadedQueue] = useAsync<TransactionListPage>(() => {
-    if (isCurrentSafe || !safeAddress) return
+    if (isCurrentSafe || !safeAddress || !isVisible) return
     return getTransactionQueue(chainId, safeAddress)
-  }, [chainId, safeAddress, isCurrentSafe])
+  }, [chainId, safeAddress, isCurrentSafe, isVisible])
```

Comment:
`isVisible` is going to be flipped back and forth as you scroll, and it will trigger the effect every time.